chromium_src/chrome/app/android/chrome_main_delegate_android.h

```diff
@@ -1,15 +0,0 @@
-/* Copyright (c) 2019 The Brave Authors. All rights reserved.
```

Comment:
@bridiver I'm removing this file because I think it is a bug to do the `#define ChromeMainDelegate BraveMainDelegate` substitution in this .h and .cc both.
Before removing this, we're still calling `ChromeMainDelegate::BasicStartupComplete` instead of `BraveMainDelegate::BasicStartupComplete` in chrome_main_delegate_android.cc.
I think if we do it in both files, when chrome_main_delegate_android.cc includes chrome_main_delegate_android.h, our definition will be undefined by `#undef ChromeMainDelegate` line in the header file, so the codes in cc files won't work as what we initially expect.

Reply:
shouldn't you also remove brave_main_delegate.cc/brave_main_delegate.h ?

Reply:
or I guess just the .h file?

Reply:
@bridiver we only do one `#define ChromeContentClient BraveContentClient` in `chromium_src/chrome/app/chrome_main_delegate.h`.
We didn't do any substitution in .cc file currently, do we want to?

Reply:
If we want to change it, I can open up a quick independent PR later since it's unrelated to this PR.

Reply:
it's definitely not a bug to do it in both. The `.cc` override only changes ```
void ChromeMainDelegateAndroid::PreSandboxStartup() {
  ChromeMainDelegate::PreSandboxStartup();

  // Start the sampling profiler after crashpad initialization.
  sampling_profiler_ = std::make_unique<MainThreadStackSamplingProfiler>();
}
``` to call the correct subclass. The header override is needed to ensure that it uses the correct subclass